src/wallet/Web3WalletProvider.ts

```diff
@@ -255,6 +255,84 @@ export class Web3WalletProvider {
 
 	}
 
+	async WalletConnectV2 (checkConnectionOnly: boolean) {
+
+		logger(2, 'connect Wallet Connect V2');
+
+		const walletConnectProvider = await import("./WalletConnectProvider");
+
+		const universalWalletConnect = await walletConnectProvider.getWalletConnectUniversalProviderInstance();
+
+		universalWalletConnect.on("display_uri", async (uri: string) => {
+			console.log("EVENT", "QR Code Modal open");
+	
+			QRCodeModal.open(uri, () => {
+				this.client.getUi().showError('User closed modal');
+			});
+		});
+
+		// Subscribe to session ping
+		universalWalletConnect.on("session_ping", ({ id, topic }: { id: number; topic: string }) => {
+			console.log("EVENT", "session_ping");
+		});
+  
+		// Subscribe to session event
+		universalWalletConnect.on("session_event", ({ event, chainId }: { event: any; chainId: string }) => {
+			console.log("WC V2 EVENT", "session_event");
+		});
+  
+		// Subscribe to session update
+		universalWalletConnect.on("session_update", ({ topic, session }: { topic: string; session: SessionTypes.Struct }) => {
+			console.log("WC V2 EVENT", "session_updated");
+			// setSession(session);
+		},
+		);
+  
+		// Subscribe to session delete
+		universalWalletConnect.on("session_delete", ({ id, topic }: { id: number; topic: string }) => {
+			console.log("WC V2 EVENT", "session_deleted");
+			// resetApp();
```

Comment:
I presume this is when the user deletes the WC session from the wallet. In this case we should probably delete the connection and show the wallet selector UI.

Reply:
@snowwhitedev @nicktaras I've added the logic to disconnect on our end when we receive this event, however there is a bug in the WC provider at the moment that prevents this event from firing.